pkg/natdiscovery/natdiscovery_suite_test.go

```diff
@@ -80,27 +81,53 @@ func parseProtocolResponse(buf []byte) *natproto.SubmarinerNatDiscoveryResponse
 	return response
 }
 
-func createTestListener(endpoint *types.SubmarinerEndpoint) (*natDiscovery, chan []byte) {
+func createTestListener(endpoint *types.SubmarinerEndpoint) (*natDiscovery, chan []byte, chan *NATEndpointInfo) {
 	listener, err := newNatDiscovery(&types.SubmarinerEndpoint{Spec: endpoint.Spec})
+	Expect(err).To(Succeed())
+
 	readyChannel := make(chan *NATEndpointInfo, 100)
 	listener.SetReadyChannel(readyChannel)
-	Expect(err).NotTo(HaveOccurred())
+	Expect(err).To(Succeed())
```

Comment:
just opinion comment:

The error should not have "Succeeded", newNatDiscovery yes..., sounds weird to me for an error.

Also I'm spanish Succeeds sounds very similar to "Happened" (sucedido) which I guess it makes it weirder for me but that's a different topic.

For those reasons I always tend to like more .NotTo(HaveOccurred()).

Reply:
I used to do `NotTo(HaveOccurred())` when switched to `To(Succeed())` 😄  It's a matter of taste.